src/engine/ScreenShotTaker.cs

```diff
@@ -5,7 +5,7 @@
 /// <summary>
 ///   Singleton handling screenshot taking
 /// </summary>
-public class ScreenShotTaker : Node
+public class ScreenShotTaker : InputNode
```

Comment:
I'll do a full review later, but this name started bothering me a few days ago when I saw it.
I think "InputNode" give the wrong connotation, that this is just something for input handling. I'd prefer a name like NodeWithInput, that way it is clear that this is an expanded Node class "with input".